### PR TITLE
Rename `_insecure_md5` to `md5`

### PR DESCRIPTION
**Pull request by harupy**
### Related Issues/PRs

 #10106

### What changes are proposed in this pull request?

I started thinking `_insecure_md5` sounds strange. To me, it sounds like we're implementing an insecure version of MD5, but it already is insecure. Let me know if there is a better name.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/__init__.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def get_results_from_paginated_fn(paginated_fn, max_results_per_page, max_result
     return all_results
 
 
-def _insecure_md5(string=b""):
+def _md5_hash(string=b""):
```
Comment (harupy) [inline mlflow/utils/__init__.py:275]:
```suggestion
def _hash(name, string=b""):
```

We might want to add `name` for #10106.

Comment (B-Step62) [reply mlflow/utils/__init__.py]: What about `_insecure_hash`? I see see value in keeping explicit alert in its name, and if we no longer say "md5" anyway then the original concern ("is it insecure version of md5?") won't be a big matter.

**Review by B-Step62 — APPROVED** [review]
LGTM!

File: mlflow/utils/__init__.py (hunk 1 above)
Comment (harupy) [reply mlflow/utils/__init__.py]: I added a `insecure_hash` module instead :)

**Comment by harupy** [comment]
@B-Step62 Could you add SHA1 to `insecure_hash.py` to close #10106?

**Comment by B-Step62** [comment]
Sure thing, will do soon!

**Comment by harupy** [comment]
@B-Step62 Thanks! We should also consider enabling https://docs.astral.sh/ruff/rules/hashlib-insecure-hash-function/ to prevent using insecure hashing algorithms directly.

**Pull request merged**